parser/ini_test.go

```diff
@@ -0,0 +1,436 @@
+package parser
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+const file_path = "./testdata/sample.ini"
+const wrong_file_path = "./testdata/wrong_sample.ini"
+
+func TestReadFromString(t *testing.T) {
```

Comment:
should cover more scenarios e.g
- empty file
- key = (missing value)
- comments on the same key value line